### DISCUSSION TITLE
Why assert is used over throwing an exception?

### DISCUSSION THREAD

**Discussion by markus-fischbacher** [nlohmann/json#3668] [Q&A] [answered]
Hi there, 
just a short question:
Why is assert (JSON_ASSERT) used to check preconditions over throwing an exception?
Assert causes a hard exit. No chance to catch and keep the application running.

Regards,
Markus

**Comment by falbrechtskirchinger** [comment, marked as answer]
Did you read the relevant section in the [documentation](https://json.nlohmann.me/features/assertions/)?

Are you asking because of a specific assertion?

**Reply by markus-fischbacher** [reply]
@falbrechtskirchinger thanks! I've read the documentation. Especially the Integration/Compiler/CMake section. Not the "small printed" stuff at the end.
It was just a question about assert overall. Not a specific one.
The NDEBUG define was missing in my CMake release configuration. It's obviously not added automatically.
